frontend/src/editor/player.tsx

```diff
@@ -191,40 +167,189 @@ export function PlayerBar({ documentId, editor }: { documentId: string; editor:
 
       <div className="pb-24" />
     </>
   );
 }
 
+function BufferingPlayButton({
+  onClick,
+  playing,
+  buffering,
+}: {
+  onClick: () => void;
+  playing: boolean;
+  buffering: boolean;
+}) {
+  const timeoutRef = useRef<NodeJS.Timeout>();
+  const [delayedBuffering, setDelayedBuffering] = useState(false);
+
+  useEffect(() => {
+    if (buffering) {
+      timeoutRef.current = setTimeout(() => {
+        setDelayedBuffering(true);
+      }, 500);
+    } else {
+      if (timeoutRef.current) {
+        clearTimeout(timeoutRef.current);
+        timeoutRef.current = undefined;
+      }
+      setDelayedBuffering(false);
+    }
+  }, [buffering]);
+
+  return (
+    <IconButton
+      icon={playing ? (delayedBuffering ? LoadingSpinner : ImPause2) : ImPlay3}
+      label={playing ? 'pause' : 'play'}
+      size={28}
+      onClick={onClick}
+    />
+  );
+}
+
+function SeekBar({
+  time,
+  duration,
+  onSeek,
+  editor,
+}: {
+  time: number;
+  duration: number | undefined;
+  onSeek: (time: number) => void;
+  editor: Editor;
+}) {
+  const ref = useRef<HTMLDivElement>(null);
+  const [dragging, setDragging] = useState(false);
+  const [tmpTime, setTmpTime] = useState<number | undefined>();
+
+  const onMouseSeek = useCallback(
+    (e: MouseEvent) => {
+      if (!ref.current || !duration) return;
+      const boundingRect = ref.current.getBoundingClientRect();
+      const seekToPercent = (e.pageX - boundingRect.x) / boundingRect.width;
+      const clampedPercent = Math.max(0, Math.min(seekToPercent, 1.0));
+
+      onSeek(clampedPercent * duration);
+      setTmpTime(undefined);
+    },
+    [duration],
+  );
+
+  const onMouseMove = useCallback(
+    (e: MouseEvent | React.MouseEvent) => {
+      if (!ref.current || !duration) return;
+      const boundingRect = ref.current.getBoundingClientRect();
+      const seekToPercent = (e.pageX - boundingRect.x) / boundingRect.width;
+      const clampedPercent = Math.max(0, Math.min(seekToPercent, 1.0));
+
+      setTmpTime(clampedPercent * duration);
+    },
+    [duration],
+  );
+
+  const onMouseDown = useCallback(
+    (e: MouseEvent | React.MouseEvent) => {
+      onMouseMove(e);
+      setDragging(true);
+    },
+    [onMouseMove],
+  );
+
+  useEffect(() => {
+    if (!dragging) return;
+
+    window.addEventListener('mousemove', onMouseMove);
+
+    const onMouseUp = (e: MouseEvent) => {
+      onMouseSeek(e);
+      setDragging(false);
+    };
+    window.addEventListener('mouseup', onMouseUp);
+
+    return () => {
+      window.removeEventListener('mousemove', onMouseMove);
+      window.removeEventListener('mouseup', onMouseUp);
+    };
+  }, [dragging, onMouseSeek]);
+
+  const speakerColors = useContext(SpeakerColorsContext);
+
+  return (
+    <div
+      ref={ref}
+      className={clsx('relative', 'h-6', 'flex', 'items-center', 'select-none')}
+      onMouseDown={onMouseDown}
+    >
+      <div className="absolute h-2 w-full overflow-hidden rounded-md bg-gray-600">
```

Comment:
```suggestion
      <div className="absolute h-2 w-full overflow-hidden rounded-md bg-gray-200 dark:bg-gray-600">
```

In light mode, gray-600 looks like it could be a speaker color instead of just the default backgound

Reply:
Oh yes, that's way to dark